### PR TITLE
Run unit test with gradle command

### PR DESCRIPTION
**Pull request by zhangyihui1**

### CODE REVIEW

File: test/scripts/run_test.py [hunk 1]
```diff
@@ -47,47 +48,51 @@ def analyse_instrumentation_test_result(result):
     return ok_num
 
 
-def analyse_unit_test_result(result):
+def analyse_unit_test_result(result, mode, module):
+    xml_file_name = module + '-' + mode.value + '-result-' + XML_SUFFIX
     total_num = 0
-    fail_num = 0
+    success_num = 0
+    p = re.compile(".*?XML test result file generated at (.*?)\. Total tests (\d+).*?passed (\d+)")
```
Comment (hankzh89) [inline test/scripts/run_test.py]: This is already a structural xml file, why are you still using regular express on it?

File: test/scripts/run_test.py [hunk 2]
```diff
@@ -26,15 +26,16 @@
                              'target': 'oms.conference'},
               'p2p': {'path': os.path.join(HOME_PATH, 'src/sdk/p2p'), 'target': 'oms.p2p'}}
 
-LOGCAT_SUFFIX = str(int(time.time())) + '.log'
-
+TIMESTAMP =  str(int(time.time()))
+LOGCAT_SUFFIX = TIMESTAMP + '.log'
+XML_SUFFIX = TIMESTAMP + '.xml'
```
Comment (hankzh89) [inline test/scripts/run_test.py]: `XML_SUFFIX` isn't a good name from my perspective.

File: test/scripts/run_test.py [hunk 3]
```diff
@@ -47,47 +48,51 @@ def analyse_instrumentation_test_result(result):
...
+            m = re.match(p, line)
+            if m is not None:
+                shutil.copyfile(m.group(1), os.path.join(os.path.split(result)[0], xml_file_name))
+                total_num = int(m.group(2))
+                success_num = int(m.group(3))
+    return total_num, success_num
 
 
-def exec_am_cmd(am_cmd, device, target_package, result_file, logcat_file):
+def exec_cmd(cmd, device, target_package, result_file, logcat_file, test_path):
```
Comment (hankzh89) [inline test/scripts/run_test.py]: i assume that `test_path` is a path that the `cmd` will be executed.
if i am right, then `test_path` is not an appropriate name

File: test/scripts/run_test.py [hunk 4]
```diff
@@ -47,47 +48,51 @@ def analyse_instrumentation_test_result(result):
...
-            exec_am_cmd(am_cmd, device, target_package, result_file, logcat_file)
+            exec_cmd(adb + am_cmd, device, target_package, result_file, logcat_file, test_path)
     elif mode == TestMode.UNIT:
-        am_cmd = ['shell', 'am', 'instrument', '-w', '-r', '-e', 'debug', 'false',
-                  target_package + '.test/android.support.test.runner.AndroidJUnitRunner']
-        exec_am_cmd(am_cmd, device, target_package, result_file, logcat_file)
+        if device is not None:
+            os.environ['ANDROID_SERIAL'] = device
+        cmd = [HOME_PATH + '/gradlew', '-q', 'clean']
+        subprocess.call(cmd, cwd=test_path)
```
Comment (hankzh89) [inline test/scripts/run_test.py]: i believe there is a `exec_cmd` method to do this job.

File: test/scripts/run_test.py (hunk 1 above)
Comment (hankzh89) [reply test/scripts/run_test.py]: then the whole PR is unnecessary .........

Comment (zhangyihui1) [reply test/scripts/run_test.py]: done

File: test/scripts/run_test.py (hunk 2 above)
Comment (zhangyihui1) [reply test/scripts/run_test.py]: done

File: test/scripts/run_test.py (hunk 3 above)
Comment (zhangyihui1) [reply test/scripts/run_test.py]: done

File: test/scripts/run_test.py (hunk 4 above)
Comment (zhangyihui1) [reply test/scripts/run_test.py]: done

File: test/scripts/run_test.py [hunk 5]
```diff
@@ -47,47 +49,58 @@ def analyse_instrumentation_test_result(result):
     return ok_num
 
 
-def analyse_unit_test_result(result):
-    total_num = 0
-    fail_num = 0
+def analyse_unit_test_result(result, mode, module):
```
Comment (hankzh89) [inline test/scripts/run_test.py]: it's better take a `result_file.xml` as this function's parameter

File: test/scripts/run_test.py [hunk 6]
```diff
@@ -47,47 +49,58 @@ def analyse_instrumentation_test_result(result):
...
+    xml_file_name = module + '-' + mode.value + '-result-' + TIMESTAMP + '.xml'
+    p = re.compile(".*?XML test result file generated at (.*?\.xml)")
```
Comment (hankzh89) [inline test/scripts/run_test.py]: the result file from `gradlew cAT` will be output to a specific path which you don't need to get it through a regular expression

File: test/scripts/run_test.py [hunk 7]
```diff
@@ -47,47 +49,58 @@ def analyse_instrumentation_test_result(result):
...
+    error_num = int(test_suite.attributes["errors"].value)
+    success_num = total_num - failed_num - error_num
+    return total_num, success_num
+
+
+def exec_cmd(mode, cmd, device, target_package, result_file, logcat_file, cmd_path):
+    if device is not None:
+        os.environ['ANDROID_SERIAL'] = device
+    if mode == TestMode.UNIT:
+        subprocess.call([HOME_PATH + '/gradlew', '-q', 'clean'], cwd=cmd_path)
```
Comment (hankzh89) [inline test/scripts/run_test.py]: No need to `gradlew clean` for instrumentation test?

File: test/scripts/run_test.py [hunk 8]
```diff
@@ -135,13 +148,11 @@ def run_unit_test(log_dir, device):
     result = True
     for module, unit_test in UNIT_TESTS.items():
         prepare_unit_test(unit_test['path'])
-        install_test(unit_test['path'], device)
-        result_file = run_cases(TestMode.UNIT, module, unit_test['target'], log_dir, device)
-        total, failure = analyse_unit_test_result(result_file)
-        succeed = total - failure
+        result_file = run_cases(TestMode.UNIT, module, unit_test['target'], log_dir, device, unit_test['path'])
```
Comment (hankzh89) [inline test/scripts/run_test.py]: `result_file` is useless for unit test.

File: test/scripts/run_test.py [hunk 9]
```diff
@@ -47,47 +49,58 @@ def analyse_instrumentation_test_result(result):
...
+        subprocess.call([HOME_PATH + '/gradlew', '-q', 'clean'], cwd=cmd_path)
     adb = ['adb'] if device == None else ['adb', '-s', device]
     clean_logcat = ['logcat', '-c']
     subprocess.call(adb + clean_logcat)
     with open(result_file, 'a+') as rf:
-        subprocess.call(adb + am_cmd, stdout=rf)
+        subprocess.call(cmd, cwd=cmd_path, stdout=rf ,stderr=rf)
     logcat_cmd = ['logcat', '-d', target_package]
     with open(logcat_file, 'a+') as lf:
         subprocess.call(adb + logcat_cmd, stdout=lf)
```
Comment (hankzh89) [inline test/scripts/run_test.py]: This function is named as `exec_cmd` which, i assume, means to execute a command basically. So running a logcat here doesn't seem clear and logical.

File: test/scripts/run_test.py (hunk 8 above)
Comment (hankzh89) [reply test/scripts/run_test.py]: And please check how to run `gradlew cAT` on a specific module. After changing to using gradle command, the `target` or `path` probably won't be necessary any more.

File: test/scripts/run_test.py (hunk 6 above)
Comment (zhangyihui1) [reply test/scripts/run_test.py]: I/XmlResultReporter: XML test result file generated at  (report path/connected/TEST-SM-G9500 - 7.0-p2p-.xml). Total tests 3, failure 1, passed 2,  脚本获取xml reoport 比用配置文件的方式更加合理,1: 如果device 改变,不需要修改脚本,2 : 如果跑2个device 不需要增加一个配置和修改脚本

File: test/scripts/run_test.py (hunk 7 above)
Comment (zhangyihui1) [reply test/scripts/run_test.py]: instrumentation test aolso do this, but not in same method

File: test/scripts/run_test.py (hunk 8 above)
Comment (zhangyihui1) [reply test/scripts/run_test.py]: run_cases is a General method for unit test and instrumentation test , target need for instrumentation , and result_file  is recorded logcat for unit test and instrumentation test

File: test/scripts/run_test.py (hunk 5 above)
Comment (zhangyihui1) [reply test/scripts/run_test.py]: done

File: test/scripts/run_test.py (hunk 9 above)
Comment (zhangyihui1) [reply test/scripts/run_test.py]: done

**Pull request closed without merging**